### DISCUSSION TITLE
Extremely strange compile behaviour

### DISCUSSION THREAD

**Discussion by dmk978** [winnow-rs/winnow#601] [Q&A] [answered]
I have a project with dependency:

```
winnow = "^0"
```

EVERY time I bump a version of MY project (just change one number in package.version), it fails to compile as if it suddenly downgrades winnow to previous version.

For example, compiler states that

```
error::ContextError::new().add_context(...)
```

should have 2 arguments instead of 3, as it was several months ago.

Removing Cargo.lock helps to build project again until new version bump.

Why?

**Comment by epage** [comment, marked as answer]
This is sounding kind of like rust-lang/cargo#14613 or one of the issues linked off of it.

FYI Cargo recommends *against* using version ranges like `"^0"` ([source](https://doc.rust-lang.org/nightly/cargo/reference/specifying-dependencies.html#multiple-requirements)).

**Reply by dmk978** [reply]
Thanks, more precise version designation helped.

Actually, I had 3 crates that switched versions in the way like winnow.
